code/_onclick/hud/map_popups.dm

```diff
@@ -0,0 +1,86 @@
+/client/
+	var/list/screen_maps = list() //assoc list with all the active maps - when a screen obj is added to a map, it's put in here as well. "mapname" = list(screen objs in map)
+
+/obj/screen
+	var/assigned_map = null
+	var/list/screen_info = list()//x,x pix, y, y pix || x,y
+
+/client/proc/clear_all_popups(var/map_to_clear = null)//not really needed most of the time, as the client's screen list gets reset on relog. any of the buttons are going to get caught by garbage collection anyway. they're effectively qdel'd.
+	if(!map_to_clear)
+		return FALSE
```

Comment:
Wouldn't it make more sense to just remove all of them if no arg is given by default?

Reply:
I've split it into two procs, one explicit "clear every popup" and one "clear x popup" so there's no situation down the line where a null somehow gets in there and fucks up everything.